web/modules/custom/nfafmis/nfafmis.views.inc

```diff
@@ -37,4 +37,68 @@ function nfafmis_views_query_alter(ViewExecutable $view, QueryPluginBase $query)
       }
     }
   }
+  elseif ($view->storage->get('id') === 'cfr_lists') {
+    // Get the current user's assigned management unit.
+    $nfafmis_service_utility = \Drupal::service('nfafmis_service.utility');
+    $range = $nfafmis_service_utility->getUserManagementUnit();
+
+    if ($range) {
+      // Add a join to the management unit data table.
+      $definition = [
+        'table' => 'taxonomy_term__management_unit',
+        'type' => 'INNER',
+        'field' => 'entity_id',
+        'left_table' => 'taxonomy_term_field_data',
+        'left_field' => 'tid',
+      ];
+
+      $join = Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      $query->addRelationship('taxonomy_term__management_unit', $join, 'person');
+
+      // Limit CFRs to those that are in the user's management unit.
+      $query->addWhere('conditions', 'taxonomy_term__management_unit.management_unit_target_id ', $range->id(), '=');
+    }
+  }
+}
+
+/**
+ * Implements hook_views_pre_render().
+ *
+ * @param \Drupal\views\ViewExecutable $view
+ */
+function nfafmis_views_pre_render(ViewExecutable $view) {
+  $view->element['#attached']['library'][] = 'nfafmis/nfafmis';
+
+  // On first load of Farmer Filter the farmer name is not set and the most
+  // recent farmer is returned. We don't want that, so don't show farmer details
+  // if the name filter was not set.
+  if ($view->id() == 'farmer_filter') {
+    if (empty($view->getExposedInput()['title'])) {
+      $result = $view->result;
+      foreach ($result as $i => $row) {
+        $view->result[$i]->_entity = NULL;
+      }
+    }
+  }
+
+  if ($view->id() == 'fmis_reports' && ($view->current_display == 'farmers_full' || $view->current_display == 'farmers_full_export')) {
+    // The farmer report view has a relationship to content with an entity
```

Comment:
Not sure what this does, but it would appear that a filter would be also a better tool for this?

Reply:
That's another old chunk of code that's been moved. Ideally, yes, it would be a filter

Reply:
I'm going to put this on the long finger for now because it's not related to the current issue. I've added a @todo comment